lang/cs/Org.Apache.REEF.Wake/Remote/AuxillaryStreamingFunctions.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    /// <summary>
+    /// Static class containing functions to read and write some basic types from/to stream
+    /// </summary>
+    public static class AuxillaryStreamingFunctions
+    {
+        /// <summary>
+        /// Read bytes from the stream.
+        /// </summary>
+        /// <param name="stream">Stream fromwhich to read</param>
+        /// <param name="bytesToRead">Number of bytes to read</param>
+        /// <param name="buffer">byte buffer to read in to</param>
+        /// <param name="offset">offset in the buffer where we should start writing</param>
+        /// <returns>The number of bytes read if success otherwise -1</returns>
+        internal static int ReadBytes(Stream stream, int bytesToRead, byte[] buffer, int offset = 0)
+        {
+            int totalBytesRead = 0;
+            while (totalBytesRead < bytesToRead)
+            {
+                int bytesRead = stream.Read(buffer, offset + totalBytesRead, bytesToRead - totalBytesRead);
+                if (bytesRead == 0)
+                {
+                    // Read timed out or connection was closed
+                    return -1;
+                }
+
+                totalBytesRead += bytesRead;
+            }
+
+            return bytesToRead;
```

Comment:
its nota bug....this function returns only if both are same. Anyways I have changedit